src/Extension/CoreExtension.php

```diff
@@ -1300,6 +1302,65 @@ function twig_test_iterable($value)
     return $value instanceof \Traversable || \is_array($value);
 }
 
+/**
+ * Checks if a variable is a sequence.
+ *
+ *    {# evaluates to true if the foo variable is a list or a traversable object #}
+ *    {% if foo is sequence %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a sequence
+ */
+function twig_test_sequence($value)
+{
+    return $value instanceof \Traversable || (\is_array($value) && \array_is_list($value));
+}
+
+
+/**
+ * Checks if a variable is a mapping.
+ *
+ *    {# evaluates to true if the foo variable is an array but not a list #}
+ *    {% if foo is mapping %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a mapping
+ */
+function twig_test_mapping($value)
+{
+    return \is_array($value) && (empty($value) || !\array_is_list($value));
```

Comment:
We tried never to use `empty`.

Reply:
Thanks, I did replace empty($value) by (0 === \count($value)).